ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywall.kt

```diff
@@ -44,9 +46,13 @@ internal fun InternalPaywall(
     options: PaywallOptions,
     viewModel: PaywallViewModel = getPaywallViewModel(options),
 ) {
+    viewModel.trackPaywallImpressionIfNeeded()
+    BackHandler {
+        viewModel.close()
```

Comment:
Is `viewModel.close` called twice now with this?

Reply:
No, the `BackHandler` is only called when the user presses the back button/performs the back navigation gesture.

We only call `close` manually when the user presses the `X` button or dismisses the PaywallDialog (by tapping outside it).

Reply:
Oh so we weren't calling that then, good catch